src/ziggurat/streams.clj

```diff
@@ -160,33 +163,97 @@
       (finally
         (.finish span)))))
 
-(defn- topology [handler-fn {:keys [origin-topic oldest-processed-message-in-s]} topic-entity channels]
-  (let [builder           (StreamsBuilder.)
-        topic-entity-name (name topic-entity)
-        topic-pattern     (Pattern/compile origin-topic)]
-    (.addStateStore builder (store-supplier-builder))
-    (->> (.stream builder topic-pattern)
+(defn- join-streams
```

Comment:
why do we still have the logic to join n streams, when we are only supporting joining of 2 streams in M1?
I say this because this is adding unnecessary complexity.

Reply:
the added complexity is only calling `reduce`, join-streams function is just a combinator...

Reply:
hmm, okay this will do for now. Do you think it makes sens to move joins specific functionality to a different namespace?

Reply:
lets keep it on streams.clj